### PR TITLE
[UN-1506] throw error iso of returning null (switchNetwork)

### PR DESCRIPTION
**Pull request by weeebr**
https://popstand.atlassian.net/browse/UN-1506

### CODE REVIEW

File: src/wallet/index.ts [hunk 1]
```diff
@@ -69,11 +69,12 @@ const registerAccountChangedListener = () => {
  * @returns address of the connected account
  */
 export const connect = async (): Promise<string | null> => {
```
Comment (tgmarinho) [inline src/wallet/index.ts:71]: the function `checkIsWalletInstalled` should throw an error.

```
export const checkIsWalletInstalled = () => {
  if (typeof window === "undefined") {
    throw new Error("Should run through Browser");
  }
  // Have to check the ethereum binding on the window object to see if it's installed
  const { ethereum } = window;

  if (ethereum && ethereum?.isMetaMask) return ethereum;

  throw new Error("No wallet installed");
};

```

**Review by tgmarinho — CHANGES_REQUESTED** [review]

File: src/wallet/index.ts (hunk 1 above)
Comment (weeebr) [reply src/wallet/index.ts:71]: why should a **check-only** function throw an error in your opinion? it's just a yes/no information you get if called... which is exactly what we need to inform the user about a lack of a web3 wallet. but have a look at https://github.com/unicrowio/sdk/pull/118, i think it's better to have it implemented that way.

Comment (tgmarinho) [reply src/wallet/index.ts:71]: to avoid if/else

**Comment by tgmarinho** [comment]
conflict

File: src/wallet/index.ts (hunk 1 above)
Comment (tgmarinho) [reply src/wallet/index.ts:71]: and handle it properly using try/catch, once the user of SDK should have web3wallet installed.

Comment (tgmarinho) [reply src/wallet/index.ts:71]: but ok let's keep this way for now, just fix the conflict and lets merge

**Pull request merged**